src/noise.cpp

```diff
@@ -83,6 +83,10 @@ u32 PcgRandom::range(u32 bound)
 	if (bound == 0)
 		return next();
 
+	// Optimization: only one value is possible.
+	if (bound == 1)
+		return 0;
```

Comment:
I would add this to `range(s32 min, s32 max)` too and IMO the comment is not needed